client/selectionmodelclient.cpp

```diff
@@ -29,11 +29,22 @@
 #include "selectionmodelclient.h"
 #include "client.h"
 
+#include <QTimer>
+
 using namespace GammaRay;
 
 SelectionModelClient::SelectionModelClient(const QString& objectName, QAbstractItemModel* model, QObject* parent) :
-  NetworkSelectionModel(objectName, model, parent)
+  NetworkSelectionModel(objectName, model, parent), m_timer(new QTimer(this))
 {
+  m_timer->setSingleShot(true);
+  m_timer->setInterval(125);
+  Q_ASSERT(model);
+  // We do use a timer to group requests to avoid network overhead
+  connect(model, SIGNAL(modelAboutToBeReset()), this, SLOT(clearPendingSelection()));
+  connect(model, SIGNAL(rowsInserted(QModelIndex,int,int)), m_timer, SLOT(start()));
+  connect(model, SIGNAL(columnsInserted(QModelIndex,int,int)), m_timer, SLOT(start()));
```

Comment:
Shouldn't layoutChanged be in this list instead, row/column changes should automatically update persistent model indexes, and thus selections, no?